servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/KtorGraphQLRequestParser.kt

```diff
@@ -61,6 +62,8 @@ open class KtorGraphQLRequestParser(
 
     private suspend fun parsePostRequest(request: ApplicationRequest): GraphQLServerRequest? = try {
         request.call.receive()
+    } catch (e: CannotTransformContentToTypeException) {
+        throw e
     } catch (e: IOException) {
         throw IllegalStateException("Invalid HTTP request - unable to parse GraphQL request from POST payload", e)
```

Comment:
Removed `InvalidPayloadException` that was added in the commit https://github.com/ExpediaGroup/graphql-kotlin/pull/2066/commits/30c32b907e12111ec949520e938143b4d3cf81b1

Instead, `CannotTransformContentToTypeException` can be re-thrown to maintain current behaviour (and pass [this test](https://github.com/ExpediaGroup/graphql-kotlin/pull/2066/files#diff-a3a141f98a8823ff7328bc1781c3f6a176f64e5c36420f824518807108b51c1bR198)).

The reason being that `ContentTransformationException` (super class of `CannotTransformContentToTypeException`) derives from `IOException` in Ktor 3.x which results in the exception being caught here in `parsePostRequest()` instead of being passed on in the chain to Ktor's [default interceptor](https://github.com/ktorio/ktor/blob/main/ktor-server/ktor-server-core/common/src/io/ktor/server/engine/BaseApplicationEngine.kt#L114) (which is where the 415 error response is returned).
Re-throwing this exception allows Ktor's internal interceptor to catch it and maintain existing behaviour.

Reference: `ContentTransformationException` in [2.3.13](https://github.com/ktorio/ktor/blob/2.3.13/ktor-server/ktor-server-core/jvmAndNix/src/io/ktor/server/plugins/Errors.kt#L67) vs [3.x](https://github.com/ktorio/ktor/blob/main/ktor-server/ktor-server-core/common/src/io/ktor/server/plugins/Errors.kt#L68).

An alternative to this could be to instead throw [UnsupportedMediaTypeException](https://api.ktor.io/ktor-server/ktor-server-core/io.ktor.server.plugins/-unsupported-media-type-exception/index.html). However, that would also require handling it in [GraphQLStatusPages](https://github.com/ExpediaGroup/graphql-kotlin/blob/master/servers/graphql-kotlin-ktor-server/src/main/kotlin/com/expediagroup/graphql/server/ktor/GraphQLStatusPages.kt#L32).